File: android/app/src/main/java/com/integreat/fetcher/FetchResult.java
package com.integreat.fetcher;

import java.util.Date;

import javax.annotation.Nullable;

public class FetchResult {
    private final String url;
    private final Date lastUpdate;
    private final boolean alreadyExisted;
    private final String errorMessage;

    public FetchResult(String url, Date lastUpdate, boolean alreadyExisted, String errorMessage) {
        this.url = url;
        this.lastUpdate = lastUpdate;
        this.alreadyExisted = alreadyExisted;
        this.errorMessage = errorMessage;
    }

    public String getUrl() {
        return url;
    }

    public Date getLastUpdate() {
        return lastUpdate;
    }

    public boolean alreadyExisted() {
        return alreadyExisted;
    }

    @Nullable
    public String getErrorMessage() {
        return errorMessage;
    }
}

File: native/android/app/src/main/java/tuerantuer/app/integreat/fetcher/FetchResult.kt
package tuerantuer.app.integreat.fetcher

import java.util.Date

class FetchResult(
    @JvmField val url: String,
    @JvmField val lastUpdate: Date,
    private val alreadyExisted: Boolean,
    private val errorMessage: String?
) {
    fun alreadyExisted(): Boolean {
        return alreadyExisted
    }

    fun getErrorMessage(): String? {
        return errorMessage
    }
}
